### PR TITLE
Fix support for the PackageType.VIRTUAL_SYSTEM enum

### PR DESCRIPTION
**Pull request by ogrisel**
While trying to use conda-lock with the latest conda and mamba on scikit-learn, I got errors due to the use of recently introduced `conda.models.enums.PackageType`.

Here is a PR to attempt to fix it while preserving backward compat.

Probably fixes: #2129, #1794, #1770.

### CODE REVIEW

**Comment by ogrisel** [comment]
Note: since some of the linked issues are really old. The commit in conda that introduced the `PackageType.VIRTUAL_SYSTEM` enum is 4 years old. So maybe it has always been an enum rather than a string and the backward compat protection in my PR is useless:

- https://github.com/conda/conda/commit/f32904bef0606fab12ee9d87380ca4201c5c4996

File: mamba/mamba/utils.py [hunk 1]
```diff
@@ -26,6 +26,19 @@
...
+try:
+    from conda.models.enums import PackageType
+except ImportError:
+    PackageType = None
+
+
+def _is_virtual_system(rec):
+    if PackageType is None:
+        # Backward compat for older conda versions
+        return rec.package_type == "virtual_system"
```
Comment (ogrisel) [inline mamba/mamba/utils.py:38]: Based on my latest comment above, I suppose that this is not needed and we could always import the `PackageType` enum instead. Let me know what you think.

**Comment by ogrisel** [comment]
I am not sure how to craft a minimal reproducer from the linked issues that wouldn't require access to the network and I am familiar with how you write tests for this project.

Would you prefer a minimal unit test? Or an integration test, for instance for the case reported in #1770 using `check_output` in `test_all.py`?

**Comment by jonashaag** [comment]
Any test is fine really. We have lots of tests that need the network.

**Comment by ogrisel** [comment]
I am having trouble setting up a dev environment for mamba following the instructions found here: https://mamba.readthedocs.io/en/latest/developer_zone/build_locally.html:

```
$ mamba env create --name mambadev --file ./mamba/environment-dev.yml
...
Installing collected packages: securesystemslib
Successfully installed securesystemslib-0.30.0

done
Unable to install package for sel(win).

Please double check and ensure your dependencies file has
the correct spelling.  You might also try installing the
conda-env-sel(win) package to see if provides the required
installer.
```

**Comment by ogrisel** [comment]
@jonashaag I have pushed a test. I am could not get my test setup to actually use my dev `mamba/mamba/utils.py` (even when following the instructions of the contributor guide), it would still use the `mamba/mamba/utils.py` from my `miniforge3/lib/python3.10/site-packages` for some reason.

Still I think the PR contents is working as intended and hopefully your CI can confirm this.

**Comment by jonashaag** [comment]
Perfect thank you! Would you mind also adding a Linux test? Could be a copy paste or a parameteized test

**Review by jonashaag — APPROVED** [review]

**Comment by ogrisel** [comment]
> Could be a copy paste or a parameterized test

I updated the test to parameterize it for Linux and Windows (in addition to macOS).

**Comment by AntoinePrv** [comment]
Thanks for this ! It should also be backported to the `1.x` branch if you want to see it a `1.5.3` release (not planned for now).

**Comment by JohanMabille** [comment]
We should fix the build first.

**Comment by ogrisel** [comment]
Indeed, I hardcoded the use of bash because the tests was originally written only for macOS. Now that it was parameterized for windows, we need to unhardcode the shell. I will do a follow-up PR.

**Comment by ogrisel** [comment]
@jonashaag @AntoinePrv @JohanMabille hopefully #2924 should fix the Windows CI.

**Pull request merged**